net/fetcher_test.go

```diff
@@ -84,3 +94,71 @@ func TestFetchNotBlockFormat(t *testing.T) {
 	require.Error(t, err)
 	require.Nil(t, blocks)
 }
+
+func TestGraphsyncFetcherHappyPath(t *testing.T) {
```

Comment:
I would call this an "integration test", of multiple graphsyncs/fetchers. We need it, but we also need unit tests for the fetcher code in isolation - see my comment in the constructor.

Also please extract to a graphsync_test.go file, to match moving the implementation.